### PR TITLE
Support conversion of InstructionValue to PhiValue

### PR DESCRIPTION
**Pull request by kalzoo**
## Description

This change supports the conversion of `InstructionValue` to `PhiValue` when the underlying instruction is in fact a `phi` instruction. It is a naive approach but which works in practice in the use case I have for it (which is currently private but which will be public in the next few weeks).

I wrote it following the conventions/patterns already used in the library, namely that it panics when a condition is unmet. I'd rather make it `try_as_phi_instruction` and return a `Result`, but that pattern isn't in use here.

Comments welcome; if this approach is worthy of merging I can add tests.

## Related Issue

Closes https://github.com/TheDan64/inkwell/issues/274

## How This Has Been Tested

I am using it in a pre-production library to convert an InstructionValue; otherwise, no testing.

## Option\<Breaking Changes\>

All changes are additive.

## Checklist

- [x] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

`cargo clippy` emits 1155 errors unrelated to my changes.

### CODE REVIEW

File: src/values/instruction_value.rs [hunk 1]
```diff
@@ -675,6 +675,18 @@ impl<'ctx> InstructionValue<'ctx> {
...
+
+    /// Return this `InstructionValue` as a `PhiValue`.
+    /// Panics if the instruction is not a Phi instruction.
+    pub fn as_phi_value(&self) -> PhiValue<'ctx> {
+        if self.get_opcode() == InstructionOpcode::Phi {
+            unsafe {
+                PhiValue::new(self.as_value_ref())
+            }
+        } else {
+            panic!("Found {:?} but expected the PhiValue variant", self)
```
Comment (TheDan64) [inline src/values/instruction_value.rs:687]: This should be a None rather than panicing

Comment (TheDan64) [reply src/values/instruction_value.rs:687]: Or even `TryFrom` impl

Comment (dbanty) [reply src/values/instruction_value.rs]: This is switched to `TryFrom` now, can you give it another look @TheDan64? Thanks!

**Comment by TheDan64** [comment]
I like it, could you please add a test for when it passes and for when it fails?

**Comment by nilslice** [comment]
@TheDan64 - I've added a test for both expected success and failure cases. Lmk what you think, thanks!

**Review by TheDan64 — APPROVED** [review]

**Pull request merged**